sdks/python/apache_beam/ml/inference/sklearn_inference.py

```diff
@@ -94,9 +91,6 @@ class SklearnModelHandlerPandas(ModelHandler[pandas.DataFrame,
                                              BaseEstimator]):
   """ Implementation of the ModelHandler interface for scikit-learn that
       supports pandas dataframes.
-
-      NOTE: This API and its implementation are under development and
-      do not provide backward compatibility guarantees.
```

Comment:
I think @robertwb wanted to keep the implementations with named inputs marked experimental

Reply:
+1. We're moving to marking those things that are experimental at a more fine-grained level, and the named inputs should fall into that class.

Reply:
OK. I can leave this note here, but I don't really see support for pandas dataframes as something separate. Sklearn users will want it as much as numpy array support, since sklearn models are built on top of have named inputs I'm not really understanding why we would modify our support for numpy but not sklearn arrays.

Reply:
It's a question of what is likely to change. I am 99% we'll want to change the way we handle dataframes, not so sure about numpy. We could call it safe and mark both (as long as there's still enough meat in the "non-experimental" portions).

Reply:
To be more specific on why this might change - now that the batching DoFn infrastructure is in, I'd like to make the pandas sklearn implementation to leverage it. We'd move to a model where the element type is a Beam Row (with schema), and the batch type is a pandas DataFrame. As opposed to the current model where the batch type is a list of single element dataframes.

Once we do that we could pass data from the DataFrame API (under the hood a `PCollection[pd.DataFrame]`) directly to RunInference, without having to unbatch it and then batch it back up.

Reply:
yeah. I agree with this change. Let's do that as part of a separate PR.